### PR TITLE
clSetKernelArg test cases for cl_khr_command_buffer extenstion

### PR DESCRIPTION
**Pull request by shajder**
Added two test cases according to point 1.7 of issue specification:

https://github.com/KhronosGroup/OpenCL-CTS/issues/1369

### CODE REVIEW

File: test_conformance/extensions/cl_khr_command_buffer/command_buffer_set_kernel_arg.cpp [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "basic_command_buffer.h"
+#include "procs.h"
+
+#include <vector>
+#include <fstream>
+#include <stdio.h>
```
[the remark this answers is not in the document]
Comment (shajder) [reply test_conformance/extensions/cl_khr_command_buffer/command_buffer_set_kernel_arg.cpp]: done

File: test_conformance/extensions/cl_khr_command_buffer/command_buffer_set_kernel_arg.cpp [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, nullptr);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        error = clEnqueueReadBuffer(queue, out_mem, CL_TRUE, 0, data_size(),
+                                    output_data.data(), 0, nullptr, nullptr);
+        test_error(error, "clEnqueueReadBuffer failed");
+
+        error = clFinish(queue);
+        test_error(error, "clFinish failed");
```
[the remark this answers is not in the document]
Comment (shajder) [reply test_conformance/extensions/cl_khr_command_buffer/command_buffer_set_kernel_arg.cpp]: done

File: test_conformance/extensions/cl_khr_command_buffer/command_buffer_set_kernel_arg.cpp [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+        test_error(error, "Failed to create copy kernel");
+
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int SetUpKernelArgs() override
+    {
+        // do not extend buffers unless device support simultaneous use!
+        if (!simultaneous_use_support) buffer_size_multiplier = 1;
```
Comment (EwanC) [inline test_conformance/extensions/cl_khr_command_buffer/command_buffer_set_kernel_arg.cpp:71]: Is this case not already handled in the constructor with `if (simul_use) buffer_size_multiplier = 2;`?

**Review by EwanC — APPROVED** [review]

File: test_conformance/extensions/cl_khr_command_buffer/command_buffer_set_kernel_arg.cpp (hunk 3 above)
Comment (shajder) [reply test_conformance/extensions/cl_khr_command_buffer/command_buffer_set_kernel_arg.cpp]: You're right, corrected.

**Review by EwanC — APPROVED** [review]

**Review by EwanC — APPROVED** [review]

**Review by aharon-abramson — APPROVED** [review]

**Pull request merged**